### DISCUSSION TITLE
GLSL flavors

### DISCUSSION THREAD

**Discussion by sbaer** [shader-slang/slang#5726] [Q&A] [answered]
Is it possible to have slangc output specific flavors of GLSL or GLSL ES?

I am able to run slangc on a basic HelloTriangle source, but no matter what I try I only get `#version 450` GLSL files. I would like to be able to output `#version 330 core` or `#version 320 es` GLSL from slang.

**Comment by expipiplus1** [comment, marked as answer]
There's currently no way to do target different versions of GLSL or GLSL ES with Slang.

We only officially support 450, however depending on which features you use, the output can be compatible with older versions (for example abstaining from using double precision floats, or tesselation shaders).

If there are any minor issues preventing the shaders from being used, then we can certainly take a look (or accept PRs), but any major implementation for older GLSL versions isn't currently on the roadmap.

I've never had to do it, but one other option might be to output spirv and then use spirv-cross to generate older glsl, obviously ymmv with that :)

**Reply by sbaer** [reply]
Thanks; that's unfortunate as this technology seems so promising but doesn't fill the reality that we have to deal with. This seems like a non-starter for us to start using slang.
